objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestUploadAndDeleteQuery.java

```diff
@@ -0,0 +1,612 @@
+            }
+        }
+    }
+    
+    @Test
+    public void testImageDateDeleteBefore() {
+        Map<String, Object> request = new LinkedHashMap<>();
+        request.put("imageDir", OUTPUT_DIR);
+        request.put("imageDate", START_DATE);
+        request.put("dateRange", "before");
+        
+        core.deleteLocalImages(request, null);
+        
+        File d = new File(OUTPUT_DIR);
+        File[] dList = d.listFiles();
+        
+        assert dList.length == 3;
+        
+        for (File imageFile: dList) {
+            if (imageFile.getName().equals(START_DATE + ".jpg") || imageFile.getName().equals("2019-02-05--02-35-10.jpg")) {
+                fail("Image should have been deleted locally.");
+            }
+        }
+    }
+    
+    @Test
+    public void testImageDateDeleteAfter() {
+        Map<String, Object> request = new LinkedHashMap<>();
+        request.put("imageDir", OUTPUT_DIR);
+        request.put("imageDate", START_DATE);
+        request.put("dateRange", "after");
+        
+        core.deleteLocalImages(request, null);
+        
+        File d = new File(OUTPUT_DIR);
+        File[] dList = d.listFiles();
+        
+        assert dList.length == 1;
+        assert dList[0].getName().equals("2019-02-05--02-35-10.jpg");
+    }
+    
+    @Test
+    public void testImageDateDeleteRange() {
+        Map<String, Object> request = new LinkedHashMap<>();
+        List<String> dateRange = new ArrayList<String>();
+        dateRange.add(START_DATE);
+        dateRange.add(END_DATE);
+        request.put("imageDir", OUTPUT_DIR);
+        request.put("imageDate", dateRange);
+        
+        core.deleteLocalImages(request, null);
+        
+        File d = new File(OUTPUT_DIR);
+        File[] dList = d.listFiles();
+        
+        assert dList.length == 2;
+        
+        for (File imageFile: dList) {
+            if (!imageFile.getName().equals("2019-02-05--02-35-10.jpg") && !imageFile.getName().equals("2019-02-05--02-35-22.jpg")) {
+                fail("Images should have been deleted locally.");
+            }
+        }
+    }
+    
+    // ================================================= Helper functions =================================================
+    
+    public void checkUploadToVantiq(String name) {
+        vantiqResponse = vantiq.selectOne("system.documents", name);
+        if (vantiqResponse.hasErrors()) {
+            List<VantiqError> errors = vantiqResponse.getErrors();
+            for (int i = 0; i < errors.size(); i++) {
+                if (errors.get(i).getCode().equals(NOT_FOUND_CODE)) {
+                    fail("Image should have been uploaded to VANTIQ");
+                }
+            }
+        }
+    }
+    
+    public void checkNotUploadToVantiq(String name) {
+        vantiqResponse = vantiq.selectOne("system.documents", name);
+        if (vantiqResponse.isSuccess()) {
+            fail("Image should not have been uploaded to VANTIQ");
+        }
+    }
+}
```

Comment:
I may have missed it, but I don't see any tests of the actual queries, operations, invalid things, etc.  Seems like we could use the SDK to define procedures & execute them to verify that we got what we expect (or errors when we don't expect).  As these things get more & more complicated, better to have a fuller test system.

Reply:
Almost done with fixes for all the other comments. For this addition to the tests, I was going to add it as part of the fixes for Issue #123 . Let me know if it's alright to add those tests all together, or if I should add them now.